tracing-subscriber/tests/subscriber_filters/option.rs

```diff
@@ -0,0 +1,45 @@
+use super::*;
+use tracing::Collect;
+
+// This test is just used to compare to the tests below
+#[test]
+fn just_layer() {
+    let info = subscriber::named("info")
+        .run()
+        .with_filter(LevelFilter::INFO)
+        .boxed();
+
+    let collector = tracing_subscriber::registry().with(info);
+    assert_eq!(collector.max_level_hint(), Some(LevelFilter::INFO));
+}
+
+#[test]
+fn layer_and_option_layer() {
+    let info = subscriber::named("info")
+        .run()
+        .with_filter(LevelFilter::INFO)
+        .boxed();
+
+    let debug = subscriber::named("debug")
+        .run()
+        .with_filter(LevelFilter::DEBUG)
+        .boxed();
+
+    let mut collector = tracing_subscriber::registry().with(info).with(Some(debug));
+    assert_eq!(collector.max_level_hint(), Some(LevelFilter::DEBUG));
+
+    let error = subscriber::named("error")
+        .run()
+        .with_filter(LevelFilter::ERROR)
+        .boxed();
+    // None means the other layer takes control
+    collector = tracing_subscriber::registry().with(error).with(None);
+    assert_eq!(collector.max_level_hint(), Some(LevelFilter::ERROR));
+}
+
+#[test]
+fn just_option_layer() {
+    // Just a None means everything is off
+    let collector = tracing_subscriber::registry().with(None::<ExpectSubscriber>);
+    assert_eq!(collector.max_level_hint(), Some(LevelFilter::OFF));
+}
```

Comment:
There isn't a filter impl is there? I can move these tests out of this directory

Reply:
yeah, i misspoke: there isn't a `Filter` impl for `Option<impl Filter>`, but some of these tests _do_ also exercise per-layer filtering, because they use `with_filter` on the inner subscribers. thus, those tests _do_ need to be in this test module because of feature flagging. however, we could have other tests in a different module that don't use `with_filter`, and instead use the `Layer` impl for `LevelFilter`...

Reply:
ooh, I like the `Layer` impl of `LevelFilter`, let me push that up!